server/client.go

```diff
@@ -4161,6 +4163,11 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 			} else {
 				dsubj = append(_dsubj[:0], sub.im.to...)
 			}
+
+			// Make sure deliver is set if inbound from a route.
+			if remapped && c.kind == ROUTER {
```

Comment:
What about Leaf and Gateway?

Reply:
Possible, just trying to get the fix to pass which is does and not break any existing tests.